File: Readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int A = 65;
int a = 97;

int main(void)
{
    // Prompt user for text input
    string text = get_string("Text: ");

    int letters = 0;
    int words = 1; // Start with 1 to account for the last word
    int sentences = 0;
    // Count the number of letters, words, and sentences in the text
    for (int i = 0; i < strlen(text); i++)
    {
        // Count letters
        if (isalpha(text[i]))
        {
            letters++;
        }
        // Count words
        else if (isspace(text[i]))
        {
            words++;
        }
        // Count sentences
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }

    float L = letters / (float) words * 100;
    float S = sentences / (float) words * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Grade the text based on the Coleman-Liau index
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);



    }
}

File: Python/Readability/readability.py

A = 65
a = 97

text = input("Text: ")

# Count letters, words, and sentences in the text
letter_count = 0
word_count = 1  # Start with 1 to account for the first word
sentence_count = 0

for char in text:
    if char.isalpha():
        letter_count += 1
    elif char.isspace():
        word_count += 1
    elif char in ['.', '!', '?']:
        sentence_count += 1

# Calculate the Coleman-Liau index

L = (letter_count / word_count) * 100
S = (sentence_count / word_count) * 100
index = 0.0588 * L - 0.296 * S - 15.8
round_index = round(index)

# Print the grade level based on the index
if round_index < 1:
    print("Before Grade 1")
elif round_index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {round_index}")
